sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -355,6 +355,7 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
   override def nullable: Boolean = true
 
   private lazy val constFormat: UTF8String = right.eval().asInstanceOf[UTF8String]
+  private var formatter: SimpleDateFormat = _
```

Comment:
Not sure this is worth changing, but `...: DateFormat = null`? I actually am not familiar with the convention for when to use _ vs null. I thought = _ was mostly for defs.